lib/IRGen/GenExistential.cpp

```diff
@@ -1445,7 +1445,7 @@ static const TypeInfo *createExistentialTypeInfo(IRGenModule &IGM, CanType T) {
     T = existential->getConstraintType()->getCanonicalType();
   }
   llvm::StructType *type;
-  if (isa<ProtocolType>(T))
+  if (isa<ProtocolType>(T) || isa<ParameterizedProtocolType>(T))
     type = IGM.createNominalType(T);
```

Comment:
Add a FIXME here or an assert, since this is wrong; the parametrized protocol type does not share metadata with the protocol type